Sources/Nimble/Matchers/BeGreaterThan.swift

```diff
@@ -32,10 +32,10 @@ public func > (lhs: Expectation<NMBComparable>, rhs: NMBComparable?) {
 
 #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
 extension NMBObjCMatcher {
-    @objc public class func beGreaterThanMatcher(_ expected: NMBComparable?) -> NMBObjCMatcher {
-        return NMBObjCMatcher(canMatchNil: false) { actualExpression, failureMessage in
+    @objc public class func beGreaterThanMatcher(_ expected: NMBComparable?) -> NMBPredicate {
```

Comment:
Ok so from what I can tell `NMBObjCMatcher` and `NMBPredicate` are separate classes that both conform to `NMBMatcher` — so it should be non-source-breaking to replace `NMBMatcher` with `NMBPredicate` (more specific return type), but it's technically breaking to replace `NMBObjCMatcher` with `NMBPredicate`, because they don't share a common base class other than `NSObject`. (Let me know if I'm missing something.)

It's probably unlikely to be a big deal, but worth considering.

Reply:
Hmm it's unfortunate that the `NMBIbjcMatchet` are exposed here instead of `NMBMatcher`. I'm okay to exclude them in this PR.